src/components/ContentSidebar/AddTaskButton.js

```diff
@@ -0,0 +1,86 @@
+// @flow
+import * as React from 'react';
+import Button from 'box-react-ui/lib/components/button';
+import Modal from 'box-react-ui/lib/components/modal/Modal';
+import { FormattedMessage } from 'react-intl';
+import TaskForm from './ActivityFeed/task-form';
+import type { TaskFormProps } from './ActivityFeed/task-form';
+
+type AddTaskButtonProps = {
+    isDisabled: boolean,
+};
+
+// These are used by the wrapped form
+type PassThroughProps = {
+    createTask: any,
+    getApproverWithQuery: any,
+    getMentionWithQuery: any,
+    approverSelectorContacts: any,
+    mentionSelectorContacts: any,
+    getAvatarUrl: any,
+};
+
+type Props = AddTaskButtonProps & PassThroughProps;
+
+type State = {
+    isTaskFormOpen: boolean,
+};
+
+class AddTaskButton extends React.Component<Props, State> {
+    state = {
+        isTaskFormOpen: false,
+    };
+
+    static defaultProps = {
+        isDisabled: false,
+    };
+
+    handleClickAdd = () => this.setState({ isTaskFormOpen: true });
+
+    handleClose = () => this.setState({ isTaskFormOpen: false });
+
+    handleSubmit = () => this.setState({ isTaskFormOpen: false });
+
+    render() {
+        const { isDisabled, ...passThrough } = this.props;
+        const { isTaskFormOpen } = this.state;
+        const { handleClickAdd, handleClose, handleSubmit } = this;
```

Comment:
Minor, but the convention I've usually seen for passing handlers is `onClickAdd={this.handleClickAdd}` with `on` for the prop name and `handle` for the value.